workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.ts

```diff
@@ -29,45 +29,59 @@ interface TokensExpiration {
   refreshToken: number; // ms
 }
 
-export interface CognitoAuthenticationPluginOptions {
+interface CognitoAppClient {
   /**
-   * The Cognito domain. Follows the format: "https://\<domain prefix\>.auth.\<region\>.amazoncognito.com"
+   * The Cognito app client IDs.
    */
-  cognitoDomain: string;
-
+  clientId: string | string[];
   /**
    * The Cognito user pool ID. Follows the format: "\<region\>_\<some string\>"
    */
   userPoolId: string;
 
   /**
-   * The Cognito app client ID.
+   * The Cognito app client secret.
+   */
+  clientSecret: string;
+}
+
+interface WebUiAppClient extends Omit<CognitoAppClient, 'clientId'> {
+  /**
+   * The Cognito app client IDs.
    */
   clientId: string;
+}
 
+export interface CognitoAuthenticationPluginOptions {
   /**
-   * The Cognito app client secret.
+   * The Cognito domain. Follows the format: "https://\<domain prefix\>.auth.\<region\>.amazoncognito.com"
    */
-  clientSecret: string;
+  cognitoDomain: string;
+
+  /**
+   * Web UI Cognito App Client information.
+   */
+  webUiAppClient?: WebUiAppClient;
+
+  /**
+   * Cognito App Clients information for token validation supporting ALLOW_ADMIN_USER_PASSWORD_AUTH authentication mode.
+   */
```

Comment:
They dont necessarily need to support `ALLOW_ADMIN_USER_PASSWORD_AUTH`, these are just the clients that are also allowed to verify their access tokens

Reply:
Updated comment and property name.